### ISSUE TITLE
Licensing choice

### ISSUE THREAD

**Issue by akoehn** [aaron-siegel/puzzlecad#20]
First of all, thanks for puzzlecad.  I started using it some weeks ago and really enjoy it.

I saw that you recently changed the license to CC-by-nc-nd, which is a surprising choice for a software project. [Creative Commons recommends using a software license instead of CC for software](https://creativecommons.org/about/program-areas/software/).  Also, do you really want to disallow any commercial use and derivates?

Disallowing derivates means (depending on the interpretation) that nobody is allowed to create new puzzles with puzzlecad.  It certainly means that nobody is allowed to adapt the examples. Nobody is allowed to do pull-requests with changes to puzzlecad.

Disallowing commercial use e.g. disallows Andrew Crowell from using puzzlecad for his TICs and RIPs.

**Comment by aaron-siegel** [comment]
Thanks for the feedback. I'm glad you're enjoying puzzlecad! I am struggling with the licensing a bit. Perhaps you can help me here:

I definitely want to release puzzlecad under the least restrictive feasible license. I do want it to be usable for commercial purposes (specifically, for people to create copies of their own puzzles for sale, as Andrew has). Indeed, puzzlecad.scad still references the CC - Attribution license which is minimally restrictive, and it's cross-released on Thingiverse under CC - Attribution license. I still consider puzzlecad itself to be under CC - Attribution.

The reason I changed the license is that in addition to puzzlecad itself, the repo also contains models of copyrighted puzzle designs. Commercial use of those designs or derivatives is definitely not permitted. So we have a situation where parts of the repo need to be under a more restrictive license than others. I wasn't sure how to handle that with github. (I could always split the repo, but that would be a big pain for development.)

So in summary, my goal is to make puzzlecad as unrestricted as feasible, while protecting the rights of designers whose puzzles are modeled with it. Ideally puzzlecad would be under MIT license or LGPL or CC-Attribution or some such, and the copyrighted designs under CC-by-nc-nd. Do you know of any examples where different parts of a repo use a different license, that I could use as a model?

**Comment by aaron-siegel** [comment]
To be totally clear (and for the benefit of anyone else reading this now or in the future...)

puzzlecad is released on Thingiverse under the Creative Commons - Attribution license, which does permit derivatives and commercial uses: https://www.thingiverse.com/thing:3198014

That is independent of the github repo license. I hope to have the repo license issue sorted out shortly. But regardless of the repo license, I'm on record (as the sole copyright holder of puzzlecad.scad) as saying that if you download and use the official release on Thingiverse, then as of right now, CC - Attribution is (and has always been) the effective license.

**Comment by akoehn** [comment]
That makes sense.  I would simply get rid of the LICENSE file, add the MIT license as a comment at the top of `puzzlecad.scad`.  Add a fitting header to the scad files describing puzzles with copyright, e.g.

```
/*
 * This puzzle is was created by [Name] in [year].
 * If you want to print or modify it, please make sure you are allowed to do this.
 */
```
You can place *your* examples into the public domain so everyone can use it, or use the CC-0 license (which is essentially the same, only deals with juristictions where it is impossible to place something in the public domain, such as Germany), or use CC-by if you want to be acknowledged.
Again, you can simply add a header as above, but with e.g. `Licensed under CC-by 4.0` (optionally link to license).

Then, in the README add a section:
```
LICENSE
-------
puzzlecad itself is licensed under the MIT license.

The example puzzles are copyrighted by their respective
creators, see the respective scad files for details.

The examples in puzzlecad-examples.scad are licensed under CC-by.
```

GitHub does not impose a single license throughout a repository, it merely has buttons to initialize a default LICENSE file and it tries to guess the license(s) used in a repository.

**Comment by aaron-siegel** [comment]
I've converted puzzlecad to a two-license repo, so this issue should now be resolved. Thanks for your help with this!
